examples/learned_models/cartpole.jl

```diff
@@ -0,0 +1,121 @@
+using Distributions: Normal
+using Random
+using JLD2
+using Altro
+using BilinearControl: Problems
+
+include("edmd_utils.jl")
+
+function gencartpoleproblem(x0=[0.,0.], Qv=1.0, Rv=1.0, Qfv=1000.0, u_bnd=3.0, tf=3.0; 
+                            dt=0.05, constrained=true)
+    model = RobotZoo.Cartpole()
+    dmodel = RD.DiscretizedDynamics{RD.RK4}(model) 
+    n,m = RD.dims(model)
+    N = round(Int, tf/dt) + 1
+
+    Q = Qv*Diagonal(@SVector ones(n)) * dt
+    Qf = Qfv*Diagonal(@SVector ones(n))
+    R = Rv*Diagonal(@SVector ones(m)) * dt
+    xf = @SVector [0, pi, 0, 0]
+    obj = TO.LQRObjective(Q,R,Qf,xf,N)
+
+    conSet = TO.ConstraintList(n,m,N)
+    bnd = TO.BoundConstraint(n,m, u_min=-u_bnd, u_max=u_bnd)
+    goal = TO.GoalConstraint(xf)
+    if constrained
+        TO.add_constraint!(conSet, bnd, 1:N-1)
+        TO.add_constraint!(conSet, goal, N:N)
+    end
+
+    X0 = [@SVector fill(NaN,n) for k = 1:N]
+    u0 = @SVector fill(0.01,m)
+    U0 = [u0 for k = 1:N-1]
+    Z = TO.SampledTrajectory(X0,U0,dt=dt*ones(N-1))
+    prob = TO.Problem(dmodel, obj, x0, tf, constraints=conSet, xf=xf) 
+    TO.initial_trajectory!(prob, Z)
+    TO.rollout!(prob)
+    prob
+end
+
+mutable struct CartpoleParams 
+    x0::Product{Continuous, Uniform{Float64}, Vector{Uniform{Float64}}}
+    QRratio::Uniform{Float64}
+    Qfratio::Uniform{Float64}  # log of ratio
+    tf::Uniform{Float64}
+    u_bnd::Uniform{Float64}
+    dt::Float64
+    function CartpoleParams(;x0_bnd=[1.0,pi/2,10,10], QRratio=[0.1, 10], Qfratio=[1.0, 4.0], 
+                             tf=[4.0, 7.0], u_bnd=[2.0, 6.0], dt=0.05)
+        x0_sampler = Product([Uniform(-x0_bnd[i],x0_bnd[i]) for i = 1:4])
+        QR_sampler = Uniform(QRratio[1], QRratio[2])
+        Qf_sampler = Uniform(Qfratio[1], Qfratio[2])
+        tf_sampler = Uniform(tf[1], tf[2])
+        u_bnd_sampler = Uniform(u_bnd[1], u_bnd[2])
+        new(x0_sampler, QR_sampler, Qf_sampler, tf_sampler, u_bnd_sampler, dt)
+    end
+end
+
+function Base.rand(params::CartpoleParams) 
+    x0 = rand(params.x0) 
+    R = 1.0 
+    Q = rand(params.Qfratio)
+    Qf = 10^(rand(params.Qfratio)) * Q
+    u_bnd = rand(params.u_bnd)
+    tf_raw = rand(params.tf)
+    N = round(Int, tf_raw / params.dt) + 1
+    tf = params.dt * (N - 1)
+    (x0=x0, Qv=Q, Rv=R, Qfv=Qf, u_bnd=u_bnd, tf=tf)
+end
+
+## Visualizer
+model = RobotZoo.Cartpole()
+visdir = joinpath(@__DIR__, "../examples/visualization/")
+include(joinpath(visdir, "visualization.jl"))
+vis = Visualizer()
+open(vis)
+delete!(vis)
+set_cartpole!(vis)
+
+## Cartpole ALTRO data
+Random.seed!(1)
+model = RobotZoo.Cartpole()
+dmodel = RD.DiscretizedDynamics{RD.RK4}(model)
+num_traj = 100
+tf = 5.0
+dt = 0.05
+
+params_sampler = CartpoleParams(tf=[tf-eps(tf),tf+eps(tf)], u_bnd=[3.0, 8.0])  # restrict it to a set horizon, for now
+opts = Altro.SolverOptions(show_summary=false)
+ctrl = ALTROController(gencartpoleproblem, params_sampler, opts=opts)
+resetcontroller!(ctrl, zeros(4))
```

Comment:
This method samples new parameters, creates a new problem, and solves it using ALTRO.